client/utils.go

```diff
@@ -4,15 +4,19 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
+	"sync"
 
 	cerrdefs "github.com/containerd/errdefs"
 	"github.com/docker/docker/api/types/filters"
-	"github.com/docker/docker/internal/lazyregexp"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
-var headerRegexp = lazyregexp.New(`\ADocker/.+\s\((.+)\)\z`)
+var (
+	headerRegexpOnce sync.Once
+	headerRegexp     *regexp.Regexp
+)
```

Comment:
Using a `sync.OnceValue` may actually be cleaner here;

```suggestion
var headerRegexp = sync.OnceValue(func() *regexp.Regexp {
	return regexp.MustCompile(`\ADocker/.+\s\((.+)\)\z`)
})
```


_However_, it looks like we're overly complicating things, because we also have an `OSType` header that's sent on every response;

- Originally added in docker 17.04; https://github.com/moby/moby/pull/30788
- Updated to also being returned for "API version not supported" errors; in 17.11; https://github.com/moby/moby/pull/35151


Which means we can replace the whole `getDockerOS()` to just read the header; let me open a quick PR

Reply:
Heh, yes, indeed; I went looking and all of a sudden realised "we have another way to do this!" - much cleaner 🎉